bin/bootstrax

```diff
@@ -278,16 +278,18 @@ def kill_process(pid, wait_time=None):
         os.kill(pid, sig)
 
 
-def set_state(state, pid_process = None, run_number = None):
+def set_state(state, pid_process = None, run_number = None, hostname = None):
     """Inform the bootstrax collection we're in a different state
 
     if state is None, leave state unchanged, just update heartbeat time
     if a pid_process and run_number are provided, update the 'usage' collection
     """
-    if state_doc_id is not None and pid_process is not None and run_number is not None:
+    if (state_doc_id is not None and pid_process is not None and
```

Comment:
Perhaps it's clearer to add a new method (update_monitoring or something) and call it directly from line 544 (just under get_status). That keeps get_status simple and you don't need this big if statement anymore (which can cause confusion: if people forget one of the arguments, the usage update is silently skipped)